code/obj/flock/cage.dm

```diff
@@ -120,60 +186,121 @@
+				target = C //set target to the critter
+
 			if(target)
 				target.set_loc(src)
 		else
-			underlays -= target
 			if(hasvar(target, "health"))
 				var/absorption = min(absorb_per_process_tick, target:health)
 				target:health -= absorption
 				reagents.add_reagent(target_fluid, absorption * 2)
 				if(target:health <= 0)
-					reagents.add_reagent(target_fluid, 10)
-					qdel(target)
-					target = null
+					if(isliving(target))
+						target.set_loc(src.loc) //kick it out and gib it
+						target:gib()
+						occupant = null
+						playsound(src, "sound/impact_sounds/Flesh_Tear_2.ogg", 80, 1)
+						src.visible_message("<span class='alert bold'>[src] rips what's left of its occupant to shreds!</span>")
+					else
+						if(iscritter(target))
+							occupant = null
+							playsound(src, "sound/impact_sounds/Flesh_Tear_2.ogg", 80, 1)
+							src.visible_message("<span class='alert bold'>[src] rips what's left of its occupant to shreds!</span>")
+						target.set_loc(null)
+						qdel(target)
+						target = null
 			else
 				reagents.add_reagent(target_fluid, 10)
+				target.set_loc(null)
 				qdel(target)
 				target = null
+
+		UpdateIcon()
+
 		if(occupant)
-			underlays -= occupant
-			underlays += occupant
 			if(eating_occupant && prob(20))
 				boutput(occupant, "<span class='flocksay italics'>[pick_string("flockmind.txt", "flockmind_conversion")]</span>")
-		if(src.contents.len <= 0 && reagents.get_reagent_amount(target_fluid) < 50)
-			if(reagents.has_reagent(target_fluid)) // flood the area with our unprocessed contents
+		if(src.contents.len <= 0 && reagents.get_reagent_amount(target_fluid) < create_egg_at_fluid)
+			if(reagents.has_reagent(target_fluid)) // dump out our excess resources as a cache
 				playsound(src, "sound/impact_sounds/Slimy_Splat_1.ogg", 80, 1)
-				T.fluid_react_single(reagents.get_reagent_amount(target_fluid))
+				var/obj/item/flockcache/x = new(src.loc)
+				x.resources = reagents.get_reagent_amount(target_fluid)
+				reagents.del_reagent(target_fluid,x.resources)
 			qdel(src)
 
 	disposing()
 		playsound(src, "sound/impact_sounds/Energy_Hit_2.ogg", 80, 1)
-		processing_items -= src
-		occupant?.removeOverlayComposition(/datum/overlayComposition/flockmindcircuit)
+
+		if(istype(occupant,/mob/living))
+			var/mob/living/M = occupant
+			M?.removeOverlayComposition(/datum/overlayComposition/flockmindcircuit)
+
+		for(var/atom/movable/AM in src)
+			if(ismob(AM))
+				var/mob/M = AM
+				M.visible_message("<span class='alert'><b>[M]</b> breaks out of [src]!</span>","<span class='alert'>You break out of [src]!</span>")
+			AM.set_loc(src.loc)
 		..()
 
+	relaymove(mob/user as mob)
+		if (user.stat)
+			return
+
+		if(prob(25))
+			takeDamage(1)
+		return
+
+	takeDamage(var/damage)
+		src.health -= damage
+		if(src.health <= 0)
+			qdel(src)
+			return
+		else
+			var/wiggle = 3
+			while(wiggle > 0)
+				wiggle--
+				src.pixel_x = rand(-2,2)
+				src.pixel_y = rand(-2,2)
+				sleep(0.5)
```

Comment:
might want to SPAWN this, not sure how it's called and we don't want to accidentally hang some game loop

Reply:
will change. This is just straight copied from icecube though, so it ought to be fine. I'll fix it anyway.